lib/execution_engine2/sdk/EE2Runjob.py

```diff
@@ -258,14 +248,150 @@ def _prepare_to_run(self, params, concierge_params=None) -> JobSubmissionParamet
             source_ws_objects=params.get(_SOURCE_WS_OBJECTS),
         )
 
+    def _prepare_to_run(self, params, concierge_params=None) -> JobSubmissionParameters:
+        """
+        Creates a job record and creates the job submission params
+        """
+
+        job_id = self._init_job_rec(self.sdkmr.get_user_id(), params)
+        self.logger.debug(
+            f"User {self.sdkmr.get_user_id()} attempting to run job {params[_METHOD]} {params}"
+        )
+        return self._generate_job_submission_params(job_id, params)
+
+    def _run_multiple(self, runjob_params):
```

Comment:
So now there are two job running paths: `_run` and `_run_multiple`, and `_run_multiple` can do everything that `_run` can do. It seems to me we could simplify the code by getting rid of `_run` and have everything use the `_run_multiple` code path.

Reply:
Right, but then we lose the more detailed error message handling for single run jobs

Reply:
Ok, how about making a ticket to look into it further?

I would be interested in knowing how much we're degrading the error messaging by switching `run_job_batch` to use `_run_multiple` vs `_run` though

Reply:
I think this question is still open yeah?

Reply:
Created a ticket https://kbase-jira.atlassian.net/browse/DATAUP-537

Reply:
Ok, but for the purposes of this PR i'd like to know how the error messaging is going to change for batch jobs, since we're actually making the change right now

Reply:
* If jobs fail the update to queued, instead of a mongoengine error, it will be a pymongo error. Kafka messages will be sent only after all updates are done, not after each update. 
* If jobs fail after add_child_jobs, it throws a slightly different mongoengine error instead
* Currently the entire job lifecycle is done before a submission fails, So create, submit, update to queued, then eventually a submission failure, followed by terminating all jobs. The new way to fail is to fail in between each of these lifecycle stages. For batch, I don't think it makes much of a difference right now.

After looking more closely, I think the error messaging will only significantly change when condor is unavailable, and when job submission is in a new thread, and the cell will transition to terminated state after clicking "Run" or "Retry" and then the cell will have to wait for the narrative to update the next job status.

Reply:
Ok, so am I right in thinking that the error messaging changes are basically just from an error that your average user isn't going to be able to understand / do anything about to a different error with the same properties?

Reply:
Probably. 
My guess would be though in the scenario that if condor is down, and the cell crashes right away, the user will try a few times and give up.. But if the cell crashes later, they might keep submitting the jobs since they see the cell transition to a queued state for a bit.

Reply:
Ok. IMO then we should move stuff that uses _run to _run_multiple, but that doesn't need to happen in this PR.